### ISSUE TITLE
Mapping doesn't work if two start with the same letter

### ISSUE THREAD

**Issue by carlevison** [NeoVintageous/NeoVintageous#308]
### Description

I have two separate mappings one which is a single letter the other two letters, I can't get mapping to work which contains two letters as it triggers the first one instantly.

### Steps to Reproduce

Create two separate mappings for eg:  one called <leader>f and another called <leader>fn

### Expected behaviour

Both mappings to work

### Actual behaviour

The mapping <leader>f runs every time without having time to trigger the other mapping.

### Versions

NeoVintageous v1.5.1

Sublime Dev 3157

MacOS 10.13.3

**Comment by gerardroche** [comment]
In Vim if you define two mappings with the same prefix, and press the prefix, it waits a moment and if another key isn't pressed it executes the mapping. For example:

```
nnoremap f 3w
nnoremap fb 3b
```

Pressing `fb` will execute `3b`, and if you press just `f` and wait, then `3w` is executed.

In NeoVintageous (nv), the current behaviour just executes the the mapping if it matches.  

Disambiquating the mappings works:

```
nnoremap fw 3w
nnoremap fb 3b
```

The feature can be implemented, but it's not necesarily trivial. We'd have to set some kind of callback timer when a partial mapping is found, and then we'd have to make sure that no race condition mangles any operations.

Would you prefer the current behaviour to emit an error rather than execute the command *if* the matched mapping is only a partial mapping? For example, if your example, instead of executing the `f` mapping it would print an error message in the console and do nothing?

**Comment by carlevison** [comment]
Yeah, I think displaying an error message in the console and then do nothing would be the best bet. 

I will change the mapping for now so they don't conflict.

**Comment by eugenesvk** [comment]
In addition to the timer another good check is see if any non-matching key `c` is pressed after `f`, then you can execute the original `f` before having to wait for a timer to end since you know it's not going to be `fb` (think that's also how NeoVim does it)
